energy-integration/fees-collector/src/additional_locked_tokens.rs

```diff
@@ -0,0 +1,54 @@
+use common_types::{Nonce, Week};
+
+elrond_wasm::imports!();
+
+#[elrond_wasm::module]
+pub trait AdditionalLockedTokensModule:
+    crate::config::ConfigModule
+    + crate::fees_accumulation::FeesAccumulationModule
+    + crate::events::FeesCollectorEventsModule
+    + week_timekeeping::WeekTimekeepingModule
+{
+    #[only_owner]
+    #[endpoint(setLockedTokensPerBlock)]
+    fn set_locked_tokens_per_block(&self, locked_tokens_per_block: BigUint) {
+        self.accumulate_additional_locked_tokens();
+        self.locked_tokens_per_block().set(locked_tokens_per_block);
+    }
+
+    fn accumulate_additional_locked_tokens(&self) {
+        let last_update_week_mapper = self.last_locked_token_add_week();
+        let last_update_week = last_update_week_mapper.get();
+        let current_week = self.get_current_week();
+        if last_update_week == current_week {
```

Comment:
We do not want to terminate the function, as this part only applies if we need to distribute the tokens for the next week. Otherwise, it means that the tokens were already distributed and the function should proceed to allow the user to claim his tokens.